src/pages/[id].page.jsx

```diff
@@ -29,20 +29,28 @@ export default function Details({ details }) {
   const [htmlCode, setHtmlCode] = useState('');
   const [cssCode, setCssCode] = useState('');
   const [javascriptCode, setJavascriptCode] = useState('');
+  const [canRenderSandpack, setCanRenderSandpack] = useState(false);
 
   // dynamic imports need to be asynchronous
   useEffect(() => {
-    const updateCodeBlocks = async () => {
-      const html = await import(`../code-examples/${slug}/html.js`);
-      setHtmlCode(html.default);
+    const updateSandpackExamples = async () => {
+      try {
+        const html = await import(`../code-examples/${slug}/html.js`);
+        setHtmlCode(html.default);
 
-      const css = await import(`../code-examples/${slug}/css.js`);
-      setCssCode(css.default);
+        const css = await import(`../code-examples/${slug}/css.js`);
+        setCssCode(css.default);
 
-      const javascript = await import(`../code-examples/${slug}/javascript.js`);
-      setJavascriptCode(javascript.default);
+        const javascript = await import(
+          `../code-examples/${slug}/javascript.js`
+        );
+        setJavascriptCode(javascript.default);
+        setCanRenderSandpack(true);
+      } catch (e) {
+        setCanRenderSandpack(false);
```

Comment:
NITPICK: I would add a check to see if the value is set to true before setting it to false here. I say this because if the value is already false, we will be forcing a render that is not needed.
```
canRenderSandpack && setCanRenderSandpack(false);
```

Reply:
Ok got it. So that bit is saying if `canRenderSandpack` is true, reset it to false?